src/components/smart/Bluetooth.js

```diff
@@ -164,7 +257,17 @@ export default class Bluetooth extends Component {
      * @returns {Promise<void>}
      */
     async disconnect(id){
-        throw new Error("Bluetooth: disconnect from device not implemented")
+        BleManager.disconnect(id)
+            .then(() => {
+
+                console.log('Disconnected');
+            })
+            .catch((error) => {
+                // Failure code
+                console.log(error);
+            });
+        BleManager.removePeripheral(id).then('Device was removed')
```

Comment:
This can cause an error because disconnect is async. If the peripheral is removed before it's disconnected an error may appear...

0.5/1